### ISSUE TITLE
[API-Gateway] API Gateway throws "Invalid permissions on Lambda function"

### ISSUE THREAD

**Issue by jitsunen** [aws/aws-cdk#9617]
API Gateway created through CDK keeps throwing 
`Execution failed due to configuration error: Invalid permissions on Lambda function`

### Reproduction Steps
Snippet from provision-stack.js.

```
createLambda(environment, id, path) {
    return new lambda.Function(this, id, {
      code: new lambda.AssetCode(path),
      handler: 'index.handler',
      runtime: lambda.Runtime.NODEJS_12_X,
      environment: environment
    });
  }

  createApiGateway(lambdaHandler, path) {
    const api = new apigateway.RestApi(this, path + 'Api', {
      restApiName: path + "API",
      endpointConfiguration: {
        types: [apigateway.EndpointType.REGIONAL]
      }
    });
    const resource = api.root.addResource(path);
    const proxyResource = resource.addProxy({
      defaultIntegration: new apigateway.LambdaIntegration(lambdaHandler, {
        credentialsPassthrough: true,
      }),
      defaultCorsPreflightOptions: {
        allowCredentials: true,
        allowHeaders: ["Content-Type", "X-Amz-Date", "Authorization", "X-Api-Key", "X-Amz-Security-Token", "X-Amz-User-Agent"],
        allowMethods: ["OPTIONS", "GET", "PUT", "POST"],
        allowOrigins: ["customdomain.com"]
      },
      defaultMethodOptions: {
        authorizationType: apigateway.AuthorizationType.IAM
      }
    });
    return proxyResource;
  }
```

### What actually happened?

CloudWatch logs for API Gateway show

```
2020-08-11T18:34:02-07:00] (49a8171c1ffa39511525368c2871b821) (ba8131a7-1c52-4f28-be85-6adba38542e9) Extended Request Id: RIgJmFyKvHcFqIg=
[2020-08-11T18:34:02-07:00] (49a8171c1ffa39511525368c2871b821) (ba8131a7-1c52-4f28-be85-6adba38542e9) Execution failed due to configuration error: Invalid permissions on Lambda function
[2020-08-11T18:34:12-07:00] (06f3d4031b4a2f37ae90b895b35fdaa0) (8a9d65d5-0147-44f4-9e83-ef00458fa302) Extended Request Id: RIgLQGbFvHcFR1A=
[2020-08-11T18:34:12-07:00] (06f3d4031b4a2f37ae90b895b35fdaa0) (8a9d65d5-0147-44f4-9e83-ef00458fa302) Execution failed due to configuration error: Invalid permissions on Lambda function
```

If I create the API Gateway with lambda proxy integration manually it works fine.
I also noticed that the proxy resource's method response says "Select an integration response" when created through CDK, whereas when created manually it says "HTTP Status: Proxy". Screenshot from AWS Console attached.

![APIGateway](https://user-images.githubusercontent.com/7054704/89965923-d59b6e00-dc02-11ea-9dad-465ae99b1aff.jpg)

### Environment

  - **CLI Version      : 1.57.0**
  - **Framework Version:**
  - **Node.js Version:** v12.14.1
  - **OS               :** Ubuntu
  - **Language (Version):** Javascript

### Other

--- 

This is :bug: Bug Report

**Comment by nija-at** [comment]
My guess is that something is not working here in relation to the IAM authorizer you've configured. 

For IAM authorizer to work, an IAM policy needs to be attached. Which credentials and how are you invoking the Rest API endpoint?

**Comment by jitsunen** [comment]
My setup is using Cognito. When a user signs in, depending on their cognito group they assume an IAM role. 

```
const staffRole = new iam.Role(this, APP + '-StaffGroupRole', {
      assumedBy: new iam.WebIdentityPrincipal('cognito-identity.amazonaws.com'),
      inlinePolicies: {
        'staffApiInvoke': new iam.PolicyDocument({
          statements: [new iam.PolicyStatement(
            {
              actions: ['execute-api:Invoke'],
              effect: iam.Effect.ALLOW,
              resources: [staffProxy.anyMethod.methodArn]
            }
          ),
            new iam.PolicyStatement(
              {
                actions: ['execute-api:Invoke'],
                effect: iam.Effect.ALLOW,
                resources: [adminProxy.anyMethod.methodArn]
              }
            )]
        })
      }
    });
```

Here adminProxy and staffProxy are objects returned by the createApiGateway method. The client is a ReactJS SPA which uses AWS Amplify library to call AWS API Gateway. Manually provisioned API Gateway works with the client side code so I doubt there is anything broken on the client side. 

Do you have an example of API Gateway provisioned through CDK using IAM Authorizer and Lamdba Integration which I could use?

Thanks.

**Comment by nija-at** [comment]
Unfortunately, we don't have examples for IAM or Cognito based authorizers.

Your set up is very specific to your application so it's hard to tell what about this set up is not working correctly. For us to help you, you will need to narrow down specifically which parts of the configuration is not working.
One way to go about this is to strip down the application to a point where it works and then slowly start adding all of the configuration back in and checking what causes the error.

**Comment by jitsunen** [comment]
Its been more than a month since I ran into this problem and reported it. Although I would like to help debug the issue, unfortunately, my current project using CDK is behind schedule and I wont be able to. 

I will probably have to switch to using Terraform or Serverless framework instead of CDK. The reason I started using CDK was to avoid hand-coding Cloudformation template. I will take another look at CDK some time in the future.

Thanks for your help.
